viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { TreeNode, TreeNodeComponent, defaultTreeNode } from './tree-node';
+import { Message } from './message';
+
+type FilterTreeProps = {
+    nodes: TreeNode[];
+    showCheckboxes: boolean;    
+    checkedSeries: number[];                //Optional
+    onChecked: (ids: number[]) => void;     //Optional
+}
+
+type FilterTreeState = {
+    treeNodes: TreeNode[];
+}
+
+export class FilterTree extends React.Component<FilterTreeProps, FilterTreeState> {
+    static defaultProps: Partial<FilterTreeProps> = {
+        checkedSeries: [],
+        onChecked: () => {},
+    }
+    
+    constructor(props: FilterTreeProps) {
+        super(props);
+        this.state = {
+            treeNodes: this.props.nodes,
+        }
+    }
+
+    static getDerivedStateFromProps = (nextProps: FilterTreeProps, prevState: FilterTreeState) => {
```

Comment:
So this function is really necessary? why?

From this blog post, it appears it should be used sparingly https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html 
So if you think this is one of the use cases where it is needed, maybe you can add a comment to explain why.

Reply:
As discussed, the solution to just keep the Expand in FilterTree will work. But imho, I think it's better to keep it in the same place with the tree itself, since the whole thing is a state, it makes sense for them to be together :)

In that case, the function to update Expand will need to be held the parent component. That function will later be passed as props to FilterTree for updating.